components/ProgressBars/index.js

```diff
@@ -0,0 +1,138 @@
+import {
+  CircularProgressbar,
+  CircularProgressbarWithChildren,
+  buildStyles,
+} from "react-circular-progressbar";
+import { ProgressbarContainer } from "./ProgressBar.styled";
+import { useState, useEffect } from "react";
+
+export default function ProgressBars({
+  todaysKcal,
+  todaysCarbs,
+  todaysProtein,
+  todaysFat,
+  goal,
+}) {
+  const [kcalProgress, setKcalProgress] = useState(todaysKcal);
+  const [carbsProgress, setCarbsProgress] = useState(todaysCarbs);
+  const [proteinProgress, setProteinProgress] = useState(todaysProtein);
+  const [fatProgress, setFatProgress] = useState(todaysFat);
+
+  useEffect(() => {
+    setKcalProgress(todaysKcal);
+    setCarbsProgress(todaysCarbs);
+    setProteinProgress(todaysProtein);
+    setFatProgress(todaysFat);
+  }, [todaysKcal, todaysCarbs, todaysProtein, todaysFat]);
+
+  return (
+    <ProgressbarContainer>
+      <CircularProgressbarWithChildren
+        value={kcalProgress}
+        maxValue={goal.kcal}
+        circleRatio={0.5}
+        strokeWidth={10}
+        styles={{
+          root: {},
+          path: {
+            stroke: "var(--color-fg-kcal)",
+
+            strokeLinecap: "round",
+            transition: "stroke-dashoffset 1s ease 0.3s",
+            transform: "rotate(-0.25turn)",
+            transformOrigin: "center center",
+          },
+
+          trail: {
+            stroke: "var(--color-bg-progressbars)",
+
+            strokeLinecap: "round",
+            transform: "rotate(-0.25turn)",
+            transformOrigin: "center center",
+          },
+        }}
```

Comment:
These styles are repeated within each of the progress bars. 
For brevity and comprehension you can put all the shared styles into an object. Then use it in each progress bar component.

```suggestion
     const sharedStyles = {
  path: {
    stroke: "var(--color-fg-progress)",
    strokeLinecap: "round",
    transition: "stroke-dashoffset 1s ease 0.3s",
    transform: "rotate(-0.25turn)",
    transformOrigin: "center center",
  },
  trail: {
    stroke: "var(--color-bg-progressbars)",
    strokeLinecap: "round",
    transform: "rotate(-0.25turn)",
    transformOrigin: "center center",
  },
};
```

Reply:
managed to make the progressbars component look a little tidier but thats about it 
hat to define a style object for each progressbars buildStyle object because i couldn't find out how to change just the color for the respective progressbar